config/routes.rb

```diff
@@ -112,11 +112,12 @@
   get "/relation/:id" => "browse#relation", :id => /\d+/, :as => :relation
   get "/relation/:id/history" => "browse#relation_history", :id => /\d+/, :as => :relation_history
   get "/changeset/:id" => "browse#changeset", :as => :changeset, :id => /\d+/
-  get "/changeset/:id/comments/feed" => "changeset_comments#index", :as => :changeset_comments_feed, :id => /\d*/, :defaults => { :format => "rss" }
+  get "/changeset/:id/comments/feed" => "changeset_comments#feed", :as => :changeset_comments_feed, :id => /\d*/, :defaults => { :format => "rss" }
   resources :notes, :path => "note", :only => [:show, :new]
 
   get "/user/:display_name/history" => "changesets#index"
   get "/user/:display_name/history/feed" => "changesets#feed", :defaults => { :format => :atom }
+  get "/user/:display_name/history/comments" => "changeset_comments#index", :as => :history_comments
```

Comment:
I think we should use `changeset_comments` as the route name. I know that, for hysterical raisins, we use `history` in the URL but the controller is called `changeset_comments`.

Reply:
We should try to avoid the `:as` setting entirely. As we move to more resourceful routing, it becomes less and less necessary. Generally we want to use resourceful routing, and accept the auto generated route names as-is.

I need to think about the controller structure for this PR. I know that (ignoring the API) we've currently got a flat hierarchy of controllers, but this approach has disadvantages. For example, it's not clear that `app/controllers/changeset_comments_controller.rb#index` would be a user-scoped subset of changeset comments, and it would clash with any implementation of site-wide changeset_comments#index (e.g. for moderation). I want to consider the pros and cons of having a `user/changeset_comments` controller but I need to play around with that.

In any case, I would ask for resourceful routing here.

Reply:
Renamed to `changeset_comments`. But there's also `changeset_comment` path.

Reply:
For "think about the controller structure ... would clash with any implementation of site-wide ... (e.g. for moderation)" you can create another controller when that clash happens, see for example #4969.